### DISCUSSION TITLE
Standalone Avalonia Application

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#11267] [Q&A]
Hello, I'm not sure whether the question has already been asked/answered. But I'm wondering if there is any way to deploy a standalone application with Avalonia. I want to make a custom installer, and I'd like to use it for the UI. I know that we can use Native AOT to do which I already tried... and the application just crashed after opening it.

Thanks!

**Comment by maxkatz6** [comment]
Self hosted publishing + single file installation is usually what you need. AOT is relatively new alternative solution. 

https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/overview?tabs=cli

> I know that we can use Native AOT to do which I already tried... and the application just crashed after opening it.

You need to work on these issues, if you want NativeAOT specifically.

**Reply by ghost** [reply]
I figured out it came from DLLs that the app can't find, so I tried to run it from the publish folder directly, and `Avalonia.OpenGL` threw an exception "DLL not found".

Next, I tried to run the executable on a folder with all dlls and stuff but `SkiaSharp` threw the same exception.

**Comment by kuiperzone** [comment]
Not sure quite clear whether you are referring to a way to deploy your dotnet app, but if you are, try this:

https://github.com/kuiperzone/PupNet-Deploy

Will give you AppImage, Flatpak, Deb and RPM and Linux, and Setup on Windows
